reference/halyard/high-availability.md

```diff
@@ -0,0 +1,148 @@
+
+__Important:__ Please note that Halyard only supports this functionality for a [distributed Spinnaker deployment](/setup/install/environment/#distributed-installation) configured with a [manifest based Kubernetes account](/setup/install/providers/kubernetes-v2/).
+
+## HA Clouddriver
+
+ <div class="mermaid">
+ graph TB
+
+ clouddriver(Clouddriver) --> clouddriver-caching(Clouddriver-Caching);
+ clouddriver --> clouddriver-rw(Clouddriver-Read-Write);
+ clouddriver --> clouddriver-ro(Clouddriver-Read-Only);
+
+ classDef default fill:#d8e8ec,stroke:#39546a;
+ linkStyle default stroke:#39546a,stroke-width:1px,fill:none;
+
+ classDef split fill:#42f4c2,stroke:#39546a;
+ class clouddriver-caching,clouddriver-ro,clouddriver-rw,echo-scheduler,echo-slave split
+ </div>
+
+ {% include mermaid %}
+
+Clouddriver benefits greatly from isolating its operations into separate services. To split Clouddriver for increased availability, run:
+
+```hal config deploy ha clouddriver enable```
+
+When Spinnaker is deployed with this flag enabled, Clouddriver will be deployed as three different services, each only performing a subset of the base Clouddriver's operations. Although by default the three Clouddriver services will communicate with the global Redis (all Spinnaker services speak to this Redis) provided by Halyard, it is recommended that the logical Clouddriver services be configured to communicate with an external Redis service. To be most effective, the Clouddriver-Read-Only should be configured to speak to a Redis read replica, while the other two should be configured to speak to the master. This is handled by automatically by Halyard if the user provides the two endpoints using this command:
+
+```hal config deploy ha services clouddriver edit --redis-master-endpoint $REDIS_MASTER_ENDPOINT --redis-slave-endpoint $REDIS_SLAVE_ENDPOINT```
+
+More information on Redis replication can be [found here](https://redis.io/topics/replication).
+
+### Clouddriver-Caching
+
+The first of the three logical Clouddriver services is the Clouddriver-Caching service, or `clouddriver-caching`. This service is responsible for the caching and retrieving of cloud infrastructure data. Since this is all that the Clouddriver-Caching is doing, there is no communication between this service and any other Spinnaker service.
+
+This service's name when [configuring its sizing](/reference/halyard/component-sizing/) is `spin-clouddriver-caching`.
+
+To add a [custom profile](/reference/halyard/custom/#custom-profiles) or [custom service settings](/reference/halyard/custom/#custom-service-settings) for this service, use the name `clouddriver-caching`.
+
+### Clouddriver-Read-Write
+
+The second logical Clouddriver service is the Clouddriver-Read-Write service, or `clouddriver-rw`. This service handles all mutating operations aside from what the Clouddriver-Caching service does. This service can be scaled to handle an increased number of writes.
```

Comment:
Similarly here, do we need a plain-English-hyphenated name and a code-font name? If we do, that's fine, and this sentence is fine, but it's simpler if we just call them `clouddriver-rw`, `clouddriver-caching`, `clouddriver-ro`.